### ISSUE TITLE
Port-forward failureMode can be overridden based on input addresses' order

### ISSUE THREAD

**Issue by roycaihw** [kubernetes/kubernetes#71017]
https://github.com/kubernetes/kubernetes/pull/46517 enables kubectl listening on a list of user-specified addresses. The feature also defines failure modes on address binding:

- For localhost addresses (ipv4 and ipv6), silently fail if one of the two addresses fails
- For other input addresses, fail the command if any of the addresses fails

**What happened**:

It turns out that the failureMode can be overridden based on the order of input addresses https://github.com/kubernetes/kubernetes/pull/70931#pullrequestreview-174655049:

- Input addresses: `"localhost", "127.0.0.1", "::1"`, the failureModes would be
```
{protocol: "tcp4", address: "127.0.0.1", failureMode: "any"},
{protocol: "tcp6", address: "::1", failureMode: "any"},
``` 
- Input addresses: `"127.0.0.1", "::1", "localhost"`, the failureModes would be
```
{protocol: "tcp4", address: "127.0.0.1", failureMode: "all"},
{protocol: "tcp6", address: "::1", failureMode: "all"},
```

**What you expected to happen**:

I'd expect that kubectl/client-go won't take meaning from the input order. 

The corner case may be not likely to happen, but from the conversation in https://github.com/kubernetes/kubernetes/pull/46517/files#r179165644, there is still some question around how bind failure should work 1. on start and 2. on close of any address.

/kind bug

cc @goblain @lavalamp @liggitt @m1kola 
/sig api-machinery

**Comment by goblain** [comment]
I agree this should not happen, I've already discussed further work on this wth @lavalamp a bit which would also solve this issue. The idea is to actually make use of function detecting the existing loopback IPs and drop failureMode as such in favor of always failing as we would be sure we attempt to bind only to what should exist and yield success.

**Comment by m1kola** [comment]
It was resolved in #70931. So, I think, we can close it.

/close
